src/Npgsql/ThrowHelper.cs

```diff
@@ -7,6 +8,23 @@ namespace Npgsql
 {
     static class ThrowHelper
     {
+        [DoesNotReturn]
+        internal static void ThrowInvalidCastException_NotSupportedType(NpgsqlTypeHandler handler, NpgsqlParameter? parameter, Type type)
+        {
+            string? parameterName;
+
+            if (parameter is null)
```

Comment:
I like to do this kind of thing:

```suggestion
            var parameterName = parameter is null
                ? null
                : parameter.TrimmedName == string.Empty
                    ? $"${parameter.Collection!.IndexOf(parameter) + 1}"
                    : parameter.TrimmedName;
```

If you hate it I'm OK with the code as is too :)